### DISCUSSION TITLE
possible obscure ANSI bug, but not sure — custom character creation string stops working — "My pentagrams are disappearing!"

### DISCUSSION THREAD

**Discussion by ClaireCJS** [microsoft/terminal#18173] [Q&A] [answered]
So, I have a couple characters created with ANSIs via encoding of their sixel representation.

One of them I even have distilled down into an environment variable: Just echo %CREATE_PENTAGRAM% to the console and the pentagram character (a double-width simulated emoji comprised of 2 separate ANSI characters, one for left, one for right) is created.  Since I have it incorporated in my prompt I see it right away.

So.... The pentagrams are disappearing.   Funniest thing I've ever said inside the software world.

Echoing that same string stopped being able to fix it.

It's some condition I can't quite reliably simulate because I almost never notice right when it happens, even though I should.

For the life of me, I feel like it MUST be a bug for the creation string to fail to create.

But maybe there is a reason. Maybe if the character already exists, it's not supposed to re-define it? That seems ridiculous and silly, but we're talking about decisions made by DEC before most of us were born. Meanwhile I've actually owned a DEC unix box and i believe it's still my attic somewhere. But I digress.

My pentagrams are disappearing. 

Weirdest "g***amned" problem I've ever had (pun intended).

I do believe this issue was not experienced until recent upgrades of Windows Terminal, but no clue how to backtest that 

```
set CREATE_PENTAGRAM=←P0;8;1;10;0;2;20;0{ @???~}{wo_?/[MFJp@@@@@/GCA@??????/??????????;?_ow{}~???/@@@@@pJFM[/??????@ACG/??????????;??????????/??_ogKYr}{/?_ow\\NVbO/@B@???????;??????????/{}rYKgo_??/ObVN\\wo_?/???????@B@;?_OGCCAaYM/}BEIQayFAA/BKO_oMPIE/____@@AAAA@B@;YaACCGO_??/AFyaQIEB}?/IPMo_OKB?/AAA@@?????@B@;?_OG[cIQaA/}@?_OJ{CAB/BMRaAAANqa/????@@AAAB@B@;aQIc[GO_??/BC{JO_?@}?/qNAAAaRMB?/AAA@@?????←

echo %CREATE_PENTAGRAM%

set PENTAGRAM_UNCOLORED=←( @./←(B
set PENTAGRAM_RED_BLINKING=←[6m←[31m←( @./←(B←[39m←[49m←[0m←[25m←[39m←[49m←[0m

echo Here are some pentagrams: %PENTAGRAM_UNCOLORED% %PENTAGRAM_RED_BLINKING%
```

Top is right, bottom is wrong:
![image](https://github.com/user-attachments/assets/e195807b-c2db-408f-9d10-339282973c53)

Not-Really-Related Side-Challenge: To redo the sixel representation in the highest possible resolution Windows Terminal supports, for a crisper pentagram.  The trumpets i have for surrounding announcements use angles that don't antialias poorly, but the pentagram art antialiases horribly.  I can do the sixel conversion but i'm too poor an artist to do the character itself.  Fun story: I had to do the current one twice. Did it upside-down by accident so i also have PENTACLE in addition to PENTAGRAM 🤣 It's like 30 minutes, i was like "Not today, Satan!"  and flipped the proverbial table in anger at having to do it twice 😂

**Comment by j4james** [comment, marked as answer]
One possible issue I can see is that your `CREATE_PENTAGRAM` string ends with a solitary `ESC` character, when it should be the two characters `ESC` `\`. That alone shouldn't prevent the font definition from working, but it depends what follows it. If, for example, the next printable character was a `c`, that would combine with the `ESC` to produce a hard-reset sequence that would erase the font definition.

> i was like "Not today, Satan!" and flipped the proverbial table in anger at having to do it twice

If you find yourself in this situation again, my [font editor](https://github.com/j4james/vtfontmaker) has some table flipping functionality that could help.

<img src="https://github.com/user-attachments/assets/ab2a88fc-b694-467a-bfe8-7d82da9b7432" width="377" />

**Reply by ClaireCJS** [reply]
I was positive this was the reaosn, but when I put 2 escapes at the end, it is most defiitely processing that 2ⁿᵈ escape as part of the next code.

That is to say, if i make ```%CREATE_PENTAGRAM%``` end in two escapes rather than one, then 

```
echo %create_pentagram%[7m Test
```
Gives me "Test" in reverse, indicating that the 2ⁿᵈ escape combined with ``[7m`` to form ``{esc}[7m`` the reverse code and is not being parsed as part of the custom character creation.

So I think least with testing it out, 2 seems to be 1 too many, unless that, too, is a WT bug

**Reply by DHowett** [reply]
I do not believe James meant ESC ESC. I believe he meant ESC BACKSLASH.

**Reply by DHowett** [reply]
(This is my first time seeing VT Font Editor and I adore it!)

**Reply by ClaireCJS** [reply]
> I do not believe James meant ESC ESC. I believe he meant ESC BACKSLASH.

Ohhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh.

Thank you, thank you both!
